server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -77,14 +77,15 @@
     /** Rest headers that are copied to internal requests made during a rest request. */
     private final Set<RestHeaderDefinition> headersToCopy;
     private final UsageService usageService;
-    private BiFunction<Map<String,List<String>>,Boolean,Boolean> isRestCompatibleFunction;
+    private BiFunction<String, String, Version> restCompatibleFunction;
+
 
     public RestController(Set<RestHeaderDefinition> headersToCopy, UnaryOperator<RestHandler> handlerWrapper,
                           NodeClient client, CircuitBreakerService circuitBreakerService, UsageService usageService,
-                          BiFunction<Map<String, List<String>>, Boolean, Boolean> isRestCompatibleFunction) {
+                          BiFunction<String, String, Version> restCompatibleFunction) {
```

Comment:
This BiFunction and just types makes me worried that this will become less readable.
I wonder if we can create an interface for it. Will give it a go